src/ResourceManager/AzureBatch/Commands.Batch/Jobs/GetBatchJobPreparationAndReleaseTaskStatusCommand.cs

```diff
@@ -0,0 +1,78 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Batch;
+using Microsoft.Azure.Commands.Batch.Models;
+using System.Management.Automation;
+using Constants = Microsoft.Azure.Commands.Batch.Utils.Constants;
+
+namespace Microsoft.Azure.Commands.Batch
+{
+    [Cmdlet(VerbsCommon.Get, Constants.AzureBatchJobPrepAndReleaseStatus, DefaultParameterSetName = Constants.IdParameterSet),
+        OutputType(typeof(PSJobPreparationAndReleaseTaskExecutionInformation))]
+    public class GetBatchJobPreparationAndReleaseTaskStatusCommand : BatchObjectModelCmdletBase
+    {
+        [Parameter(Position = 0,
+            ParameterSetName = Constants.IdParameterSet,
+            Mandatory = true,
+            HelpMessage = "Specifies the ID of the job whose preparation and release tasks should be retrieved. You cannot specify wildcard characters.")]
+        [ValidateNotNullOrEmpty]
+        public string Id { get; set; }
```

Comment:
So I originally ahd this as a pipeline parameter too, but then I realized that the only way to get this through the pipeline really seems to be to get a `PSCloudJob` object and then apss that... and that will go via the other parameter set. The `Id` property on a different resource (say a `PSCloudPool`) would be wrong.

So I think doing it the way we have it makes sense, either pass an object from the pipeline (which includes an `Id` which we will then use) or pass the `Id` by itself.